service/history/shardContext.go

```diff
@@ -164,6 +171,7 @@ Create_Loop:
 		currentRangeID := s.getRangeID()
 		request.RangeID = currentRangeID
 		response, err := s.executionManager.CreateWorkflowExecution(request)
+		s.updateWriteBackoffLocked(err)
```

Comment:
Why can't we just wrap persistence calls with a retry policy?  Are you worried about retry within the lock?